### PR TITLE
Fix electricity calculation in case of bad meter-reading

### PR DESCRIPTION
**Pull request by shahardo**
#701

### CODE REVIEW

File: negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php [hunk 1]
```diff
@@ -51,6 +51,20 @@
...
+   * @var array
+   */
+  protected $lastGoodKnownReading = NULL;
+
+  /**
+   * Holds an array of references to bad meter reading.
+   *
+   * @var array
+   */
+  protected $saveBadMeterReadings = NULL;
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php]: => `badMeterReadings`

File: negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php [hunk 2]
```diff
@@ -504,66 +571,88 @@ public function processRawEntities($from_timestamp, $to_timestamp, $frequency) {
...
+    self::debugMessage("rate-type: $processed_record->rate_type, time-stamp: @time_from, sum-kwh: $processed_record->kwh", 3, $processed_record->timestamp);
+  }
 
-    // Output debug message.
-    self::debugMessage('After groupBy(rate_type)', 3);
+  /**
+   * Check if there are bad meter-readings in the array.
+   */
+  protected function hasBadMeterReadings() {
+    return count($this->saveBadMeterReadings) > 0;
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php]: `return count($this->saveBadMeterReadings);`

File: negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php (hunk 1 above)
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_normalizer/plugins/electricity_normalizer/ElectricityNormalizerBase.php]: `array()`

**Comment by shahardo** [comment]
Merged

**Pull request closed without merging**